framework/src/engine/consensus/certificate_generation/commit_pool.ts

```diff
@@ -308,6 +312,7 @@ export class CommitPool {
 				maxHeightCertified + 1,
 			);
 			nextHeight = Math.min(heightNextBFTParameters - 1, maxHeightPrecommitted);
+			nextHeight = Math.max(nextHeight, this._minCertifyHeight - 1);
```

Comment:
Moreover, I think there is a simpler and cleaner way to do this, correct me if I am wrong.

Don't make any change here, keep `selectAggregateCommit` same as before until line 323. And then just replace the while condition by:

`while (nextHeight > Math.max(maxHeightCertified, minCertifyHeight -1))`

This if `nextHeigh` is smaller than `minCertifyHeight`, then it will not even enter the while loop, since there is no point to do that.